infrastructure/code/code.go

```diff
@@ -298,6 +301,19 @@ func scanAndPersistBaseBranch(ctx context.Context, sc *Scanner, folderPath strin
 	return nil
 }
 
+func normalizeBranchName(branchName string) string {
+	normalized := strings.TrimSpace(branchName)
+	normalized = strings.ToLower(normalized)
+	normalized = strings.ReplaceAll(normalized, " ", "_")
+	reg, err := regexp.Compile(`[^a-z0-9_\-]+`)
+	if err != nil {
+		return ""
```

Comment:
In this case, probably a hash would serve? Sth like sha256 of branchname?

Reply:
I was thinking that it would be better to have the branch name in folder name. So that we could possibly recognize it later.

Reply:
actually it should always compile. 
Moved it to a global var. 
The ReplaceString will replace everything matched. It should match all chars not a-Z0-9\-